omegaconf/dictconfig.py

```diff
@@ -597,9 +584,9 @@ def _set_value(self, value: Any, flags: Optional[Dict[str, bool]] = None) -> Non
             elif isinstance(value, dict):
                 for k, v in value.items():
                     self.__setitem__(k, v)
-            else:
+            else:  # pragma: no cover
```

Comment:
Why is this not covered?

Reply:
No reason, I just moved this comment from the sentence below there because it made more sense.

Reply:
this comment says that test coverage is not considered this line as needed to be covered.
it's not a comment, it's a directive to the code coverage.
try to remove it and see if we still get 100% coverage.
if not, try to add a test that will make it get to 100%.
if it's impossible, please explain why.

Reply:
It's impossible, if it is a structured config the following code will check it.
```python
if (
     is_assign
     and is_container_annotation(target_type)
     and not is_container_annotation(value_type)
 ):
```
If it isn't a structured config it will wrap a new node anyways.

Reply:
If this code never runs, why do we need it?

Reply:
We don't. Well to have a default case could be useful sometime but now, no.

Reply:
ok. I guess we can keep it.
If there is no reason, keep the pragma unchanged.

Reply:
Okay